haskell/private/haskell_impl.bzl

```diff
@@ -58,13 +60,42 @@ def haskell_test_impl(ctx):
 def haskell_binary_impl(ctx):
     return _haskell_binary_common_impl(ctx, is_test = False)
 
-def _get_hpc_outputs(ctx, srcs_files):
+def _replace_extensions(srcs_files, extension):
     hpc_outputs = []
     for s in srcs_files:
-        filename = paths.replace_extension(s.basename, "")
-        hpc_outputs.append(filename + ".mix")
+        filename = paths.replace_extension(s.basename, extension)
+        hpc_outputs.append(filename)
     return hpc_outputs
 
+def _should_inspect_coverage(ctx, hs, is_test):
+    return hs.coverage_enabled and is_test and ctx.attr.expected_expression_coverage > 0
+
+def _wrap_binary_for_coverage(binary_path, hpc_path, tix_file_path, expected_expression_coverage, mix_file_paths):
+    return runfiles_boilerplate + """
+ERRORCOLOR='\033[1;31m'
+CLEARCOLOR='\033[0m'
+binary_path=$(rlocation {})
```

Comment:
could you use format with explicit names:
```
""" foo bar {baz} spam eggs""".format(baz=something) """
```

I prefer that for readability instead of positional arguments.

Reply:
I've solved this problem a different way, through expand_template as per Mathieu's suggestion below.